Minor improvements in framework to support  third-party modules

munezbn opened the pull request:
- Added scheduleDispatch API to TaskLoop . Now we can schedule a task to run after given timeout.
- Added a utility macro to export given class name as object creator function
- Added external directory and BUILD.gn for building thirdparty modules 
- Updated Image component class for supporting FastImage
  - Image Component will be base class for FastImage third-party component so updated Image component class accordingly.
  - Few members made protected.
  - Few member functions made as virtual
  - Instead of sending ImageSource , we will pass source URI string as argument to the local and network getImageData functions.

munezbn committed:
Minor improvments in taskloop and utils

Added helper macro to add thirdprty module creator function
Added schduled dispatch API in taskloop for scheduling task with a timeout

munezbn committed:
Minor improvement in framework

Added external directory and dummy BUILD.gn which will be used to download and build thirdparty modules.
Updated RNS_EXPORT_MODULE

shefayaparwin commented on rns_shell/platform/linux/TaskLoop.h:
```diff
@@ -27,6 +27,7 @@ class TaskLoop {
     void waitUntilRunning();
 
     void dispatch(Func fun);
+    void scheduleDispatch(Func fun, long long timeout);
```
can we have a comment to state the time unit expected

munezbn replied:
Done

munezbn committed:
Addressed review comments

Updated scheduleDispatch API parameter name and added comment to state the time unit expected.

munezbn committed:
Added helper macro for Component provider

Similar to module, added another macro for exporting a component provider object creator function.

munezbn committed:
Updated Image component to support fastImage

Image Component will be base class for FastImage third-party component so updated Image component class accordingly.
Few members made protected.
Few member functions made as virtual
Instead of sending ImageSource , we will pass source URI string as argument to the local and network getImageData functions.
